src/v/config/property.h

```diff
@@ -459,7 +459,12 @@ bool property<T>::is_nullable() const {
 
 template<typename T>
 bool property<T>::is_array() const {
-    if constexpr (detail::is_collection<std::decay_t<T>>) {
+    if constexpr (
+      std::is_same_v<T, ss::sstring> || std::is_same_v<T, std::string>) {
+        // Special case for strings, which are collections but we do not
+        // want to report them that way.
+        return false;
+    } else if constexpr (detail::is_collection<std::decay_t<T>>) {
```

Comment:
ahh is `detail::is_collection` doing something like checking for begin/end members?

Reply:
Even more primitive than that, just looking for size()